apps/api/src/app/events/services/workflow.queue.service.ts

```diff
@@ -58,17 +71,82 @@ export class WorkflowQueueService {
     this.worker.on('failed', async (job, e) => {
       await this.jobRepository.updateStatus(job.data._organizationId, job.data._id, JobStatusEnum.FAILED);
       await this.jobRepository.setError(job.data._organizationId, job.data._id, e);
+
+      const lastWebhookFilterRetry =
+        job.attemptsMade === this.DEFAULT_ATTEMPTS && e.message.includes(EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER);
+
+      if (lastWebhookFilterRetry) {
+        await this.handleLastFailedWebhookFilter(job, e);
+      }
     });
+
     this.queueScheduler = new QueueScheduler('standard', this.bullConfig);
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async handleLastFailedWebhookFilter(job: any, e: Error) {
+    await this.createExecutionDetails.execute(
+      CreateExecutionDetailsCommand.create({
+        ...CreateExecutionDetailsCommand.getDetailsFromJob(job.data),
+        detail: DetailEnum.WEBHOOK_FILTER_FAILED,
+        source: ExecutionDetailsSourceEnum.WEBHOOK,
+        status: ExecutionDetailsStatusEnum.PENDING,
```

Comment:
@p-fernandez What do you think about changing here to FAILED

```suggestion
        status: ExecutionDetailsStatusEnum.FAILED,
```

But if we do we will need to add an additional flag to the createExecutionDetails to not throw an exception in this case, because no one will catch it in this case. 

In addition on the retry logs above, I want to add another icon for new warning status. WDYT? :)

Reply:
That's a good point about setting it to FAILED. I do agree.
Now that you mention I think all the status even for previous retries should be failed.

Not following about the warning. For which situations?

Reply:
So I was thinking of introducing a new status ExecutionDetailsStatusEnum.WARNING with a new icon (we sent a ping to Shlomi about it).In my case, it will be printed on all the retries but the last one, which will be FAILED status. what do you think?

Reply:
Might be useful. Just raise it with Omer and @scopsy and @davidsoderberg. But sounds good.